### PR TITLE
Add snapshots to train.py

### PR DESCRIPTION
**Pull request by MicroPanda123**
Modified train.py script to make snapshots of checkpoints.

Added 4 new config values:
- take_snapshots (default = False) - if True, saves snapshots of checkpoints at specified conditions
- make_snapshots_on_checkpoint (default = False) - if True, snapshots will be taken every time checkpoint is also saved.
- snapshot_dir (default = 'snapshots') - specifies name of folder *inside* out_dir which will be used to keep snapshots in
- snapshot_interval (default = 500) - specifies per how much iterations it should take a snapshot.

Snapshots follow naming convention of: `ckpt-{iteration at time of taking snapshot}-{training loss}-{validation loss}.pt`.

I made this change because I was worried about overfitting the network, but at the same time not using "always_save_checkpoint" meant that it was possible for it to just not save after a looong time of training, which I would also not want to encounter.

### CODE REVIEW

**Comment by MicroPanda123** [comment]
Crap, forgot to add one thing, sorry!
